minio/signer.py

```diff
@@ -92,8 +92,13 @@ def presign_v4(method, url, credentials,
     if request_date is None:
         request_date = datetime.utcnow()
 
-    parsed_url = urlsplit(url)
     content_hash_hex = _UNSIGNED_PAYLOAD
+    for k in headers.keys():
+        if k.lower() == 'x-amz-content-sha256':
+            content_hash_hex = headers.get(k, _UNSIGNED_PAYLOAD)
+            del headers[k]
```

Comment:
The one tweak we could do here is to break when we find it, but it's not absolutely necessary. The size of headers will never be very large!

Reply:
I added the break.